server/api.js

```diff
@@ -6,6 +6,43 @@ import { v4 as uuidv4 } from "uuid";
 const itemsPerPage = 25;
 const router = Router();
 
+const auth = async (req, res, next) => {
+	try {
+		const authHeader = req.headers.authorization;
+		const token = authHeader?.split(" ")[1];
+
+		if (!token) {
+			return res.status(401).json({ code: "AUTHRIZATION_REQUIRED" });
+		}
+
+		const sessionResult = await db.query(
+			"SELECT * FROM session WHERE token = $1",
+			[token]
+		);
+		const session = sessionResult.rows[0];
+
+		if (!session) {
+			return res.status(401).json({ code: "INVALID_SESSION" });
+		}
+
+		const userResult = await db.query("SELECT * FROM users WHERE id = $1", [
+			session.user_id,
+		]);
+		const user = userResult.rows[0];
+
+		if (!user) {
+			return res.status(401).json({ code: "USER_NOT_FOUND" });
+		}
+
+		req.user = user;
+		next();
+	} catch (error) {
+		res.status(500).json({ code: "SERVER_ERROR", message: error.message });
```

Comment:
```suggestion
		res.status(500).json({ code: "SERVER_ERROR" });
```

Don't return the error message to the client. It could be a security vulnerability.